### ISSUE TITLE
Helm internally is accessing files on System32 and other places through out system.

### ISSUE THREAD

**Issue by shetesagar87** [helm/helm#9978]
Output of `helm version`:
version.BuildInfo{Version:"v3.6+unreleased", GitCommit:"e9e6b07f62e2bb5067afe79782ba3dd85efbd00e", GitTreeState:"clean", GoVersion:"go1.16.6"
Output of `kubectl version`:
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T16:31:21Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.11", GitCommit:"c18ff7bd5227b42c617cf371cf315d0423e91d5e", GitTreeState:"clean", BuildDate:"2021-05-25T17:38:04Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.21) and server (1.19) exceeds the supported minor version skew of +/-1
Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

While performing helm install --upgrade  i am facing issues which tries to access my file under system 32. 

![image](https://user-images.githubusercontent.com/64983796/127200018-77296842-a5ca-45cb-a333-0f2e1cf3068f.png)

![image](https://user-images.githubusercontent.com/64983796/127200285-edf39794-8e0d-41be-afd0-3ad99ead4acb.png)

I am pretty much new with Helm.. What exactly is causing helm to access files under system32 ??

**Comment by bacongobbler** [comment]
Can you reproduce this with a sample chart, such as the output from a `helm create`?

I also noticed you're running with a custom build of Helm. Can you please try the latest release and see if it can be reproduced using the official builds?

It's not totally clear to me why those files would be accessed.

**Comment by shetesagar87** [comment]
Thanks for the response. 

I tried `helm create` and it worked successfully. Charts.yaml with default folder structure got created.

I changed the build to a stable version. Still getting an issue while executing the command in kubernetes.

![image](https://user-images.githubusercontent.com/64983796/127390010-3591ca78-5e4a-4d33-8732-03876caf9f05.png)

The files that i am referring are having the following details : 

```
values-common.yaml

deployment:
  containerPort: 8071
  volumes:
    - name: secrets-store-inline
      csi:
        driver: secrets-store.csi.k8s.io
        readOnly: true
        volumeAttributes:
          secretProviderClass: name-provider
    - name: fs-volume
      persistentVolumeClaim:
        claimName: azurefile
  volumeMounts:
    - name: secrets-store-inline
      mountPath: "/mnt/secrets-store"
      readOnly: true
    - mountPath: "/log/path"
      name: fs-volume

probes:
  enabled: true
  livenessProbe:
    httpGet:
      path: /actuator/health/liveness
      port: http
  readinessProbe:
    httpGet:
      path: /actuator/health/readiness
      port: http

service:
  enabled: true
  port: 443
  env:
    - name: env_name
      value: env_password

configmap:
  enabled: true
  configuration:
    Configuration_key : value

secretProviderClass:
  enabled: true
  name: name-provider
  secretObjects:
    type: Opaque
    data:
      - key: userKey
        objectName: Password
  keyVaultSecrets:
    - objectName: objName
      objectType: secret
      objectVersion: ''
'''
values-dev.yaml
'''
# Default values for integration webservices.
# This is a YAML-formatted file.
# Declare variables to be passed into your templates.

image:
  repository: image_name
  tag: "tag"

deployment:
  containerPort: 8071

configmap:
  enabled: true
  configuration:
    SPRING_PROFILES_ACTIVE: swagger
    SQL_URL: SQL URL

ingress:
  enabled: true
  className: ""
  annotations:
    nginx.ingress.kubernetes.io/rewrite-target: /$2
  hosts:
    - host: host
      paths:
        - path: /path/(/|$)(.*)
          pathType: Prefix
  tls:
  - hosts:
    - host.location
    secretName: aks-ingress-tls

autoscaling:
  enabled: true
  minReplicas: 1
  maxReplicas: 1
'''

Common files contains subscription information and secrets. - Common across all components. 

Also tried creating a Hyper V and doing all the required installation over there still getting a similar error.

**Comment by bacongobbler** [comment]
I see you are running `helm upgrade --install integrationservices .` from `C:\\Users\P50040507`. I assume that's the home directory for the user "P50040507", and not the location of your Helm chart.

When you call `helm upgrade --install integrationservices .`, in layman's terms you are asking Helm to load the chart in your current directory [the dot (`.`) operator] and upgrade the `integrationservices` release with that chart.

In other words, Helm is attempting to load your home directory as a Helm chart. Hence why you are seeing these errors.

Where is the helm chart located on your system?

**Comment by bacongobbler** [comment]
Ah, that would also explain the same issues described in your first example. You ran `helm upgrade .` from c:\\windows\system32. Basically you told Helm to try loading system32 as a Helm chart. Hence the error.

Have you had a chance to read the documentation yet?

https://helm.sh/docs/

**Comment by shetesagar87** [comment]
Ohh!! Thank you so much.. I really missed that part of executing the commands from the actual folders where helm charts are present. 

A great great shout out to @bacongobbler !! - Kudos.. (y)
